src/datahike/query.cljc

```diff
@@ -1097,25 +1102,27 @@
   (->> (-collect context symbols)
        (map vec)))
 
-(defmulti q (fn [query & args] (type query)))
-
 (defmethod q clojure.lang.LazySeq [query & args]
   (q {:query query :args args}))
 
 (defmethod q clojure.lang.PersistentVector [query & args]
   (q {:query query :args args}))
 
+(defmethod q clojure.lang.PersistentList [query & args]
+  (q {:query query :args args}))
+
 (defmethod q clojure.lang.PersistentArrayMap [query-map & inputs]
-  (let [query (if (contains? query-map :query) (:query query-map) query-map)
-        query (if (string? query) (edn/read-string query) query)
-        args (if (contains? query-map :args) (:args query-map) inputs)
-        parsed-q (memoized-parse-query query)
-        find (:qfind parsed-q)
+  (let [query         (if (contains? query-map :query) (:query query-map) query-map)
+        query         (if (string? query) (edn/read-string query) query)
+        query         (if (= 'quote (first query)) (second query) query)
+        args          (if (contains? query-map :args) (:args query-map) inputs)
+        parsed-q      (memoized-parse-query query)
+        find          (:qfind parsed-q)
         find-elements (dpip/find-elements find)
-        find-vars (dpi/find-vars find)
-        result-arity (count find-elements)
-        with (:qwith parsed-q)
-        returnmaps (:qreturnmaps parsed-q)
+        find-vars     (dpi/find-vars find)
+        result-arity  (count find-elements)
+        with          (:qwith parsed-q)
+        returnmaps    (:qreturnmaps parsed-q)
```

Comment:
As far as I know, we agreed to avoid unnecessary whitespace changes like these to keep the PRs clear and to the point.  But since this is such a small PR I'd say it's ok here.

Reply:
Ok, we should anyway talk about formatting and style guide. I'll create a discussion for that.